kernel/core/datamanagement/memory.cpp

```diff
@@ -3748,94 +3748,162 @@ std::vector<mu::value_type> Memory::getIndex(size_t col, const std::vector<mu::v
+            //std::sort(catIndex1.begin(), catIndex1.end());
+            //std::sort(catIndex2.begin(), catIndex2.end());
+            std::vector<mu::value_type> intersection;
+
+            //std::set_intersection(catIndex1.begin(), catIndex1.end(), catIndex2.begin(), catIndex2.end(),
+            //                      std::back_inserter(intersection));
+
+            for(auto a : catIndex1)
+                for(auto b : catIndex2)
+                    if(a == b)
+                        intersection.push_back(a);
+
+            groupedValues[2].memArray.push_back(TblColPtr(_mem.memArray[0]->copy(VectorIndex(&intersection[0], intersection.size(), 0))));
+        }
     }
 
     // Prepare vectors for each group
-    std::vector<mu::value_type> vAvg;
-    std::vector<mu::value_type> vVar;
-    std::vector<mu::value_type> vNum;
-
-    // Get the values for each group
-    for (size_t j = 2; j < _mem.memArray.size(); j++)
+    std::vector<std::vector<mu::value_type>> vAvg_(groupedValues.size());
+    std::vector<std::vector<mu::value_type>> vVar_(groupedValues.size());
+    std::vector<std::vector<mu::value_type>> vNum_(groupedValues.size());
+    for(size_t i = 0; i < groupedValues.size(); i++)
     {
-        vAvg.push_back(_mem.avg(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(j)));
-        vNum.push_back(_mem.num(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(j)));
-        vVar.push_back(intPower(_mem.std(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(j)), 2));
+        for (size_t j = 0; j < groupedValues[i].memArray.size(); j++)
+        {
+            vAvg_[i].push_back(groupedValues[i].avg(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(j)));
+            vNum_[i].push_back(groupedValues[i].num(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(j)));
+            vVar_[i].push_back(intPower(groupedValues[i].std(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(j)), 2));
+        }
     }
 
     // Calculate the overall values
-    mu::value_type overallAvg = _mem.avg(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(2, VectorIndex::OPEN_END));
-    mu::value_type overallNum = _mem.num(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(2, VectorIndex::OPEN_END));
-    mu::value_type overallVariance;
+    mu::value_type overallAvg = _mem.avg(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(0));
+    mu::value_type overallNum = _mem.num(VectorIndex(0, VectorIndex::OPEN_END), VectorIndex(0));
 
-    // Pretend that each group contains n equal measurements
-    // to calculate the ideal overall variance
-    for (size_t i = 0; i < vAvg.size(); i++)
+    //calculate SS_Within
+    mu::value_type SS_Within;
+    for(size_t i = 0; i < vAvg_.back().size(); i++)
     {
-        overallVariance += vNum[i] * intPower(vAvg[i]-overallAvg, 2);
+        for(size_t j = 0; j < vNum_.back()[i].real(); j++)
+        {
+            mu::value_type val_j = static_cast<ValueColumn*>(groupedValues.back().memArray[i].get())->getValue(j);

```

Comment:
You are not required to use the `static_cast` here. `TableColumn::getValue()` is a `virtual` method.